samples/bluetooth/scanner/src/main.c

```diff
@@ -0,0 +1,69 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <sys/printk.h>
+#include <sys/util.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <string.h>
+
+char devices[20][100] = {};
+int pck_len = 0;
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+        int i, flag = 0;
+        for(i = 0; i < pck_len; i++) {
+                if(!(strcmp(devices[i], addr_str))) {
+                        flag = 1; // Already present
+                }
+        }
+        if(!flag) {
+                strcpy(devices[i], addr_str);
+                pck_len++;
+                printk("Device found: %s (RSSI %d) \n", addr_str, rssi);
+        }
```

Comment:
You should use the duplicate filtering feature in the controller, by supplying the `options` in the `scan_param`

Reply:
Yes, I have used the filter option.